### PR TITLE
Fix for mysql bug 74483/74933

### PR DESCRIPTION
**Pull request by colakong**
An implementation of suggested fixes for bugs:

http://bugs.mysql.com/bug.php?id=74933

http://bugs.mysql.com/bug.php?id=74483

### CODE REVIEW

File: lib/mysql/connector/network.py [hunk 1]
```diff
@@ -220,10 +220,12 @@ def recv_plain(self):
...
-            packet = bytearray(4)
-            read = self.sock.recv_into(packet, 4)
-            if read != 4:
-                raise errors.InterfaceError(errno=2013)
+            packet = bytearray()
+            while len(packet) < 4:
+                chunk = self.sock.recv(1)
+                if not chunk:
+                    raise errors.InterfaceError(errno=2013)
+                packet += chunk
```
Comment (methane) [inline lib/mysql/connector/network.py]: In typically case, `sock.recv(4)` returns 4 bytes.
Calling `socket.recv()` 4 times may drop performance significantly.

```python
packet = b''
while len(packet) < 4:
    chunk = self.sock.recv(4-len(packet))
    if not chunk:
        raise errors.InterfaceError(errno=2013)
    packet += chunk
```

Comment (colakong) [inline lib/mysql/connector/network.py]: That makes sense ^^

**Comment by colakong** [comment]
FYI, this contribution is under OCA ^^

**Comment by mysql-admin** [comment]
Hi colakong, 
Unfortunately we cannot accept this contribution as it includes code from other parties and therefor not covered by your signed OCA (see section 5 of the OCA agreement). 
Thanks for your suggestion.
==Omer

**Pull request closed without merging**